extension/src/popup/views/Settings/index.tsx

```diff
@@ -25,10 +28,14 @@ export const Settings = () => {
   const { t } = useTranslation();
   const dispatch = useDispatch();
 
+  const publicKey = useSelector(publicKeySelector);
+
   const signOutAndClose = async (e: React.FormEvent) => {
     e.preventDefault();
-    // eslint-disable-next-line
+    /* eslint-disable @typescript-eslint/await-thenable */
+    await dispatch(setLastUsedAccount({ publicKey }));
```

Comment:
A user isn't guaranteed to sign out. Their session may just timeout due to the 24h timer. 

I think maybe we should call `setLastUsedAccount` whenever a user switches account to make sure we always capture the last used account

Reply:
Nice catch, I'll make this improvement

Reply:
How about this https://github.com/stellar/freighter/pull/1706/commits/20c6118673ec72535fdf359a06fbc5ceae582b24?
It's updating the `lastUsedAccount` every time users visit the main Account screen (which also happens every time users change account).

Reply:
Nice, this looks good! You could also add a conditional to only update `lastUsedAccount` if `lastUsedAccount !== publicKey` so you're not dispatching this action if the account hasn't changed

Reply:
Humm yeah I thought about it too, but then we'd have to dispatch `loadLastUsedAccount()` every time too to fetch `lastUsedAccount` so it just seemed simpler/cleaner to overwrite it every time. Wdyt?

Reply:
I wonder if it would be better to dispatch `setLastUsedAccount` in the action to switch account instead of in this view layer? I realize that right now, every time a user switches accounts that this view re-renders but I think that coupling is loose and this dependency could be easy to miss in a refactor. Thoughts?

Reply:
Which component could we consider as freighter's "Home" screen? My line of thought was to have the code below on a main component which we know would be rendered right after login and also after users switch account, so that the `publicKeySelector` would react to that and make sure to keep the `lastUsedAccount` value updated.
```
  const publicKey = useSelector(publicKeySelector);

  useEffect(() => {
    if (publicKey) {
      dispatch(setLastUsedAccount({ publicKey }));
    }
  }, [publicKey, dispatch]);
```

If we add this code only to the switch account action I'm afraid users that never switch account wouldn't have the `lastUsedAccount` property set or updated, then in this case we'd need to `setLastUsedAccount` in 2 different places: the switch account action + login action. Which wouldn't be that bad at all, but would be 1 more place to maintain.

Wdyt?

Reply:
yeah I think what you're saying makes sense, maybe there's a different way to keep it in 1 place but also in the action layer? My thought was that it would be good to have `lastUsedAccount` be set in 1 place that guarantees it is updated whenever the active account is set to a key. My only concern with doing this in the view is that is not really tied to the action, so right now this works but if we ever add a situation where we switch accounts and don't land back on this view after, we will have to implement this in those views as well where as if we had this in the action layer, it would not matter the view that gets rendered after switching accounts.

Reply:
does `setActivePublicKey` get called in every case that we set a public key, maybe if that dispatched an action to set `lastUsedAccount` then we would not have to track that in a view?

Reply:
I like this idea, I'll take a look

Reply:
@aristidesstaffieri how about this https://github.com/stellar/freighter/pull/1706/commits/26f11ae0388f159b787c834b4e6040c622abca0e?
It's updating the `lastUsedAccount` every time user subscribes to an account, which happens exactly during login and also when user switches the active account. I've deleted the extension and tested in all scenarios, seems pretty solid!

Reply:
nice 👏 yeah that lgtm!